compiler/rustc_mir/src/interpret/intern.rs

```diff
@@ -187,6 +187,12 @@ impl<'rt, 'mir, 'tcx: 'mir, M: CompileTimeMachine<'mir, 'tcx>> ValueVisitor<'mir
                 return walked;
             }
         }
+
+        // ZSTs do not need validation unless they're uninhabited
```

Comment:
This is not validation, this is interning, so I find the comment here rather confusing.

Reply:
I think actually we can skip even uninhabited ZST, precisely because this is not validation but just looking for pointers: https://github.com/rust-lang/rust/pull/78179.